src/query/storages/fuse/src/operations/analyze/collect_ndv_source.rs

```diff
@@ -0,0 +1,386 @@
+                    if let Some(column_hlls) = &block_hll {
+                        for (column_id, column_hll) in column_hlls.iter() {
+                            merged_hlls
+                                .entry(*column_id)
+                                .and_modify(|hll| hll.merge(column_hll))
+                                .or_insert_with(|| column_hll.clone());
+                        }
+                    } else {
+                        indexes.push(idx);
+                    }
+                }
+
+                if !indexes.is_empty() && self.no_scan {
+                    self.unstats_rows += segment_info.summary.row_count;
+                    self.state = State::ReadData(None);
+                    return Ok(());
+                }
+
+                for (column_id, column_hll) in merged_hlls {
+                    self.column_hlls
+                        .entry(column_id)
+                        .and_modify(|hll| hll.merge(&column_hll))
+                        .or_insert_with(|| column_hll);
+                }
+                self.row_count += segment_info.summary.row_count;
+
+                if indexes.is_empty() {
+                    self.state = State::ReadData(None);
+                } else {
+                    assert!(self.segment_with_hll.is_none());
+                    let new_hlls = Vec::with_capacity(indexes.len());
+                    self.segment_with_hll = Some(SegmentWithHLL {
+                        segment_location,
+                        block_metas: segment_info.block_metas()?,
+                        origin_summary: segment_info.summary.clone(),
+                        raw_block_hlls: block_hlls,
+                        new_block_hlls: new_hlls,
+                        block_indexes: indexes,
+                    });
+                    self.state = State::BuildHLL;
+                }
+            }
+            State::MergeHLL => {
+                let segment_with_hll = self.segment_with_hll.as_mut().unwrap();
+                let new_hlls = std::mem::take(&mut segment_with_hll.new_block_hlls);
+                let new_indexes = std::mem::take(&mut segment_with_hll.block_indexes);
+                for (new, idx) in new_hlls.into_iter().zip(new_indexes.into_iter()) {
+                    if let Some(column_hlls) = new {
+                        for (column_id, column_hll) in column_hlls.iter() {
+                            self.column_hlls
+                                .entry(*column_id)
+                                .and_modify(|hll| hll.merge(column_hll))
+                                .or_insert_with(|| column_hll.clone());
+                        }
+                        segment_with_hll.raw_block_hlls[idx] = encode_column_hll(&column_hlls)?;
+                    }
+                }
+                self.state = State::WriteMeta;
+            }
+            _ => return Err(ErrorCode::Internal("It's a bug.")),
```

Comment:
This error message is adorable, but a bit too concise. If this ever hits production, it might be tricky to debug. Could you please add more descriptive error information?